independent-projects/tools/devtools-common/src/main/java/io/quarkus/platform/catalog/processor/CatalogProcessor.java

```diff
@@ -25,11 +31,8 @@ public static CatalogProcessor of(ExtensionCatalog catalog) {
     public static List<ProcessedCategory> getProcessedCategoriesInOrder(ExtensionCatalog catalog) {
         final Map<String, List<Extension>> extsByCategory = new HashMap<>(catalog.getCategories().size());
         for (Extension e : catalog.getExtensions()) {
-            List<String> categories = ExtensionProcessor.of(e).getCategories();
-            if (categories.isEmpty()) {
-                extsByCategory.put(UNCATEGORIZED_ID, new ArrayList<>());
-                extsByCategory.get(UNCATEGORIZED_ID).add(e);
-            }
+            List<String> categories = new ArrayList<>(ExtensionProcessor.of(e).getCategories());
+            categories.add(UNCATEGORIZED_CATEGORY.getId());
```

Comment:
Don't you need to check if `categories.isEmpty()`  before adding the `UNCATEGORIZED_CATEGORY`here?

Reply:
Some tests here would help too :)

Reply:
Very good catch, fixed